components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -198,6 +239,15 @@ private SessionDataStore() {
             sqlSelect = selectSQL;
         }
 
+        //TODO: check the requirement for configurable option
+        sqlInsertTempSTORE = SQL_INSERT_TEMP_STORE_OPERATION;
+        sqlInsertTempDELETE = SQL_INSERT_TEMP_DELETE_OPERATION;
+        temporaryCache = new ArrayList<>(Arrays.asList("AuthenticationRequestCache",
```

Comment:
Aren't we binding our logic to few of these specific caches ?. Instead what about introducing new attribute to cache config itself denoting whether it's temporary or not ? For public branch ?